include/arc_utilities/dijkstras.hpp

```diff
@@ -670,6 +670,155 @@ namespace arc_dijkstras
             }
     };
 
+    template<typename NodeValueType, typename Allocator=std::allocator<NodeValueType>>
+    class SimpleGraphAstar
+    {
+    protected:
+
+        class AstarNode
+        {
+        protected:
+
+            int64_t index_;
+            int64_t backpointer_;
+            double cost_to_come_;
+            double value_;
+
+        public:
+
+            AstarNode(const int64_t index, const int64_t backpointer, const double cost_to_come, const double value) : index_(index), backpointer_(backpointer), cost_to_come_(cost_to_come), value_(value) {}
+
+            inline int64_t Index() const { return index_; }
+
+            inline int64_t Backpointer() const { return backpointer_; }
+
+            inline double CostToCome() const { return cost_to_come_; }
+
+            inline double Value() const { return value_; }
+        };
+
+        class CompareNodeFn
+        {
+            public:
+
+                constexpr bool operator()(const AstarNode& lhs, const AstarNode& rhs) const
+                {
+                    return lhs.Value() > rhs.Value();
+                }
+        };
+
+        SimpleGraphAstar() {}
+
+    public:
+
+        typedef std::pair<std::vector<int64_t>, double> AstarResult;
+
+        static AstarResult PerformAstar(const Graph<NodeValueType, Allocator>& graph, const int64_t start_index, const int64_t goal_index, const std::function<double(const NodeValueType&, const NodeValueType&)>& heuristic_fn)
+        {
+
+            // Enforced sanity checks
+            assert(start_index >= (int64_t)0);
+            assert(start_index < (int64_t)graph.GetNodesImmutable().size());
+            assert(goal_index >= (int64_t)0);
+            assert(goal_index < (int64_t)graph.GetNodesImmutable().size());
+            assert(start_index != goal_index);
+            // Setup
+            std::priority_queue<AstarNode, std::vector<AstarNode>, CompareNodeFn> queue;
+            std::unordered_map<int64_t, std::pair<int64_t, double>> explored;
+            auto heuristic_function = [&] (const int64_t node_index) { return heuristic_fn(graph.GetNodeImmutable(node_index).GetValueImmutable(), graph.GetNodeImmutable(goal_index).GetValueImmutable()); };
+            // Initialize
+            queue.push(AstarNode(start_index, -1, 0.0, heuristic_function(start_index)));
+            // Search
+            while (queue.size() > 0)
+            {
+                // Get the top of the priority queue
+                const AstarNode top_node = queue.top();
+                queue.pop();
+                // Check if the node has already been discovered
+                const auto explored_itr = explored.find(top_node.Index());
+                // We have already been here and it was cheaper then
+                if ((explored_itr != explored.end()) && (top_node.CostToCome() >= explored_itr->second.second))
+                {
+                    continue;
```

Comment:
As written, does this continue actually do anything? Why not just negate the logic of the test, and move the contents of the "else" up?